src/main/java/com/autotune/common/data/datasource/AutotuneDatasource.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+package com.autotune.common.data.datasource;
+
+import com.autotune.common.utils.CommonUtils.AutotuneDatasourceTypes;
+
+/**
+ * AutotuneDatasource is an abstraction which needs to be implemented
+ * by every datasource type which is used in Autotune
+ *
+ * Currently Supported Implementations:
+ *  - Prometheus
+ *
+ *  The Implementation should maintain 3 variables : Name, Type and Source and
+ *  should be set or get using these API's
+ *
+ *  var name:   Type String
+ *  var type:   Type AutotuneDatasourceTypes
+ *  var source: Type String
```

Comment:
Add the actual values that we use in Prometheus here.

Reply:
Done